### PR TITLE
feat: Add support for converting auto-increment to sequence

### PR DESCRIPTION
**Pull request by asthamohta**
image

UI Changes:
- A new type of issue is added
- Src Schema shows Auto-Generated Column

Changes: 
- Auto-Increment in MySql is mapped to a new sequence which is created

### CODE REVIEW

File: cmd/data.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func validateExistingDb(ctx context.Context, spDialect, dbURI string, adminClien
 	}
 	spannerConv := internal.MakeConv()
 	spannerConv.SpDialect = spDialect
-	err = utils.ReadSpannerSchema(ctx, spannerConv, client)
+	err = utils.ReadSpannerSchema(ctx, spannerConv, client, driver)
```
Comment (manitgupta) [inline cmd/data.go:233]: Why is driver relevant to reading the spanner schema? This interface looks off to me.

**Review by manitgupta — CHANGES_REQUESTED** [review]
Bunch of interfaces are polluted it seems to be by adding `driver`. Can you add the reason why source database type is relevant to functions like `ReadSpannerSchema`, `ValidateExistingDb` etc.?

File: cmd/data.go (hunk 1 above)
Comment (asthamohta) [reply cmd/data.go:233]: I have made changes to SchemaToSpannerDDL which is eventually called inside all of these functions required the driver. Unfortunately if we look at the tree, the last place where driver info is available around 5-6 functions above the one where I need it

Comment (manitgupta) [reply cmd/data.go:233]: Please evaluate if there is a way better to make these changes without having a ripple effect change on the signatures.
We can also have a call to figure out solutions if required.

File: sources/common/infoschema.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ type InfoSchema interface {
 	GetToDdl() ToDdl
 	GetTableName(schema string, tableName string) string
 	GetTables() ([]SchemaAndName, error)
-	GetColumns(conv *internal.Conv, table SchemaAndName, constraints map[string][]string, primaryKeys []string) (map[string]schema.Column, []string, error)
+	GetColumns(conv *internal.Conv, table SchemaAndName, constraints map[string][]string, primaryKeys []string, tableId string) (map[string]schema.Column, []string, error)
```
Comment (manitgupta) [inline sources/common/infoschema.go:36]: There seems to be a table `SchemaAndName`, what is the purpose of `tableId`?

File: sources/common/infoschema.go [hunk 3]
```diff
@@ -191,6 +192,7 @@ func (is *InfoSchemaImpl) processTable(conv *internal.Conv, table SchemaAndName,
 		return t, fmt.Errorf("couldn't get foreign key constraints for table %s.%s: %s", table.Schema, table.Name, err)
 	}
 
+	table.Id = internal.GenerateTableId()
```
Comment (manitgupta) [inline sources/common/infoschema.go:195]: Where was this Id generated earlier?

Comment (asthamohta) [reply sources/common/infoschema.go]: Sorry I had to assign the value not generate it. It is generated a few line before this only in the same function

**Review by manitgupta — APPROVED** [review]

**Pull request merged**